File: 0058-length-of-last-word/0058-length-of-last-word.go
func lengthOfLastWord(s string) int {
	result := 0
	sLength := utf8.RuneCountInString(s)
	isStartCounting := false

	for i := sLength - 1; i >= 0; i-- {
		char := string(s[i])
		if char != " " && !isStartCounting {
			isStartCounting = true
			result++
			continue
		}
		if char != " " && isStartCounting {
			result++
			continue
		}
		if char == " " && isStartCounting {
			break
		}

	}

	return result
}

File: 0058-length-of-last-word/0058-length-of-last-word.ts
function lengthOfLastWord(s: string): number {
  const trimpped = s.trimEnd();
  let result = 0;

  for (let i = trimpped.length - 1; i >= 0; i--) {
    const c = trimpped.charAt(i);
    if (c === " ") {
      return result;
    }
    result++;
  }

  return result;
}

